File: Data Structures & Algorithms/online-stock-span/submission-1.java
class StockSpanner {
    Deque<int[]> stack;
    public StockSpanner() {
        this.stack = new ArrayDeque<>(10);
    }
    
    public int next(int price) {
        int span = 1;
        while (!this.stack.isEmpty() && this.stack.getLast()[0] <= price) {
            span += this.stack.removeLast()[1];
        }
        this.stack.addLast(new int[] {price, span});
        return span;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */

File: Data Structures & Algorithms/maximum-frequency-stack/submission-1.py
class FreqStack:

    def __init__(self):
        self.freq = {}
        self.stacks = [[]]

    def push(self, val: int) -> None:
        count = self.freq.get(val, 0) + 1
        self.freq[val] = count
        currVal = self.freq[val]
        if currVal == len(self.stacks):
            self.stacks.append([])
        self.stacks[currVal].append(val)
        

    def pop(self) -> int:
        val_to_pop = self.stacks[-1].pop()
        self.freq[val_to_pop] -= 1
        if not self.stacks[-1]:
            self.stacks.pop()
        return val_to_pop
        


# Your FreqStack object will be instantiated and called as such:
# obj = FreqStack()
# obj.push(val)
# param_2 = obj.pop()